### ISSUE TITLE
[red-knot] Add support for properties with deleters

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#142]
Emit a diagnostic if `del` is used on a property that has no deleter configured.

See also: existing `property.deleter` test in `properties.md`

**Comment by sharifhsn** [comment]
Closed by https://github.com/astral-sh/ruff/pull/24500

Repro:
```py
class Foo:
    @property
    def x(self) -> int:
        return 42

del Foo().x
```

ty 0.0.29:

```
All checks passed!
```

ty 0.0.30:

```
error[invalid-assignment]: Cannot delete read-only property `x` on object of type `Foo`
 --> repro.py:3:9
  |
1 | class Foo:
2 |     @property
3 |     def x(self) -> int:
  |         - Property `Foo.x` defined here with no deleter
4 |         return 42
5 |
6 | del Foo().x
  |     ^^^^^^^ Attempted deletion of `Foo.x` here
  |
```
